python/src/tasks/gene_expression.py

```diff
@@ -31,7 +31,13 @@ def compute(self):
         )
         resultR = r.json()
         result = {}
+        if not len(resultR):
+            result[genes[0]] ={
+                "message": 404,
+            }
```

Comment:
Maybe for this part, you can also expand the returned error field to be something like 

```
{
   "error": 404,
   "message": "Gene not found" 
}
```

So later you can check for the existence of `error` on the UI side, instead of checking for message?

Reply:
If I put in an error field it will break the plot with an error message and try again button. Instead I'm showing the error as a notification on top of the window and the previous plot is shown.